crates/sqlite-inproc/src/lib.rs

```diff
@@ -31,77 +33,99 @@ impl InProcConnection {
-            )
-            .map_err(|e| spin_world::sqlite::Error::Io(e.to_string()))?;
-        let rows = rows
-            .into_iter()
-            .map(|r| r.map_err(|e| spin_world::sqlite::Error::Io(e.to_string())))
-            .collect::<Result<_, spin_world::sqlite::Error>>()?;
-        Ok(spin_world::sqlite::QueryResult { columns, rows })
+        parameters: Vec<sqlite::Value>,
+    ) -> Result<sqlite::QueryResult, sqlite::Error> {
+        let connection = self.connection.clone();
+        let query = query.to_owned();
+        // Tell the tokio runtime that we're going to block while making the query
+        tokio::task::spawn_blocking(move || make_query(&connection, &query, parameters))
+            .await
+            .context("internal runtime error")
+            .map_err(|e| sqlite::Error::Io(e.to_string()))?
     }
 
-    fn execute_batch(&self, statements: &str) -> anyhow::Result<()> {
-        let conn = self.connection.lock().unwrap();
-        conn.execute_batch(statements)?;
+    async fn execute_batch(&self, statements: &str) -> anyhow::Result<()> {
+        let connection = self.connection.clone();
+        let statements = statements.to_owned();
+        tokio::task::spawn_blocking(move || {
+            let conn = connection.lock().unwrap();
+            conn.execute_batch(&statements)
+                .context("failed to execute batch statements")
+        })
+        .await?
+        .context("failed to spawn blocking task")?;
         Ok(())
     }
 }
 
+fn make_query(
+    connection: &Mutex<rusqlite::Connection>,
+    query: &str,
+    parameters: Vec<sqlite::Value>,
+) -> Result<sqlite::QueryResult, sqlite::Error> {
+    let conn = connection.lock().unwrap();
+    let mut statement = conn
+        .prepare_cached(query)
+        .map_err(|e| sqlite::Error::Io(e.to_string()))?;
+    let columns = statement
+        .column_names()
+        .into_iter()
+        .map(ToOwned::to_owned)
+        .collect();
+    let rows = statement
+        .query_map(
+            rusqlite::params_from_iter(convert_data(parameters.into_iter())),
+            |row| {
+                let mut values = vec![];
+                for column in 0.. {
+                    let value = row.get::<usize, ValueWrapper>(column);
+                    if let Err(rusqlite::Error::InvalidColumnIndex(_)) = value {
+                        break;
+                    }
+                    let value = value?.0;
+                    values.push(value);
+                }
+                Ok(sqlite::RowResult { values })
+            },
+        )
+        .map_err(|e| sqlite::Error::Io(e.to_string()))?;
+    let rows = rows
+        .into_iter()
+        .map(|r| r.map_err(|e| sqlite::Error::Io(e.to_string())))
+        .collect::<Result<_, sqlite::Error>>()?;
+    Ok(sqlite::QueryResult { columns, rows })
+}
+
 fn convert_data(
-    arguments: impl Iterator<Item = spin_world::sqlite::Value>,
+    arguments: impl Iterator<Item = sqlite::Value>,
```

Comment:
The reason I used `spin_world` qualified naming was that I got really really confused about plain `sqlite` - at any give time, was it referring to it the SQLite library, the `spin-sqlite` crate, or the `sqlite` WIT interface?  I know it is actually unambiguous, but for me the unqualified form adds a lot of cognitive load.

I appreciate though that the qualified form creates a lot of visual clutter which is its own form of cognitive load...!